adal/src/telemetry/java/com/microsoft/aad/adal/DefaultEvent.java

```diff
@@ -0,0 +1,166 @@
+
+    private static String sApplicationVersion = "NA";
+
+    private static String sClientId = "NA";
+
+    private static String sClientIp = "NA";
+
+    private static String sDeviceId = "NA";
+
+    private String mRequestId;
+
+    private int mDefaultEventCount;
+
+    DefaultEvent() {
+        mEventList = new ArrayList<>();
+
+        // Keying off Application name not being null to decide if the defaults have been set
+        if (sApplicationName != null) {
+            setEvent(EventStrings.APPLICATION_NAME, sApplicationName);
+            setEvent(EventStrings.APPLICATION_VERSION, sApplicationVersion);
+            setEvent(EventStrings.CLIENT_ID, sClientId);
+            setEvent(EventStrings.CLIENT_IP, sClientIp);
+            setEvent(EventStrings.DEVICE_ID, sDeviceId);
+            mDefaultEventCount = mEventList.size();
+        }
+    }
+
+    @Override
+    public int getDefaultEventCount() {
+        return mDefaultEventCount;
+    }
+
+    @Override
+    public void setEvent(final String name, final String value) {
+        if (TextUtils.isEmpty(name)) {
+            throw new IllegalArgumentException("Telemetry setEvent on null name");
+        }
+
+        if (value == null) {
+            return;
+        }
+
+        mEventList.add(Pair.create(name, value));
+    }
+
+    @Override
+    public List<Pair<String, String>> getEvents() {
+        return Collections.unmodifiableList(mEventList);
+    }
+
+    @Override
+    public void processEvent(final Map<String, String> dispatchMap) {
+
+        if (sApplicationName != null) {
+            dispatchMap.put(EventStrings.APPLICATION_NAME, sApplicationName);
+        }
+
+        if (sApplicationVersion != null) {
+            dispatchMap.put(EventStrings.APPLICATION_VERSION, sApplicationVersion);
+        }
+
+        if (sClientId != null) {
+            dispatchMap.put(EventStrings.CLIENT_ID, sClientId);
+        }
+
+        if (sClientIp != null) {
+            dispatchMap.put(EventStrings.CLIENT_IP, sClientIp);
+        }
+
+        if (sDeviceId != null) {
+            dispatchMap.put(EventStrings.DEVICE_ID, sDeviceId);
+        }
+    }
+
+    void setDefaults(final Context context, final String clientId) {
+
+        sClientId = clientId;
+        sApplicationName = context.getPackageName();
+        try {
+            sApplicationVersion = context.getPackageManager().getPackageInfo(sApplicationName, 0).versionName;
+        } catch (PackageManager.NameNotFoundException nnfe) {
+            sApplicationVersion = "NA";
+        }
+
+        //TODO: Getting IP will require network permissions do we want to do it?
+        sClientIp = "NA";
+
+        try {
+            sDeviceId = StringExtensions.createHash(Secure.getString(context.getContentResolver(), Secure.ANDROID_ID));
+        } catch (UnsupportedEncodingException | NoSuchAlgorithmException e) {
+            sDeviceId = "";
+        }
+
+        if (mDefaultEventCount == 0) {
```

Comment:
This was called out during the feature spec, i agree that this might be duplicates, lets revisit this in iteration2 of telemetry.

---
In reply to: [90992754](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/736#discussion_r90992754) [](ancestors = 90992754)